crates/solvers/src/infra/dex/zeroex/mod.rs

```diff
@@ -155,8 +161,8 @@ pub enum Error {
     NotFound,
     #[error("quote does not specify an approval spender")]
     MissingSpender,
-    #[error("api error code {code}: {reason}")]
-    Api { code: i64, reason: String },
+    #[error("api error code {code}: {description}")]
+    Api { code: i64, description: String },
```

Comment:
I wouldn't reuse "API" errors here - these are errors that are passed by JSON from the DEX API.

For example, this would be ParaSwap returning:
```json
{
  "error": "some well formed error message"
}
```

If this is not the case, then I would create a "Status" error variant to indicate "we got an HTTP status error, but it isn't the standard JSON error format we are expecting".

Reply:
Not sure that extra type is actually needed. We get the information we are interested in. Isn't it just an implementation detail that we were able to parse the JSON and get the underlying error the API reported?

Reply:
`Api` is a variant of the existing error type. My point was to differentiate between an error that we received over the API (so some well-formed JSON that we can parse) and one where we got a non-2xx status code with some body we can't make heads or tails of.

Reply:
I see how populating the full response string in the description may be a bit overkill (since it can be a full html page), but I do think this is an API error (just based on status code and not based on the JSON content). I can create a subtype `Error::Non200Status` just for this case, but I'm also wondering if this is overkill.

Reply:
I implemented what I meant here: https://github.com/cowprotocol/services/pull/1954

I also took the opportunity to "unify" all HTTP requests to use a shared function (that standardizes trace logging and error handling).

Reply:
Consider this code from the 0x API implementation:

```rust
        match err.code {
            100 => Self::NotFound,
            _ => Self::Api {
                code: err.code,
                reason: err.reason,
            },
        }
```

The 0x API returns JSON that looks something like this:

```json
{ "code": 1337, "reason": "some error message" }
```

With the current implementation, we would treat an HTTP status error, for example the 0x API returns `HTTP 404 Not Found` with some `body`, **as if** the API had returned some JSON like:

```json
{ "code": 404, "reason": "body" }
```

Now, what happens `codes` have meanings attached to them, and they overlap with HTTP status codes? We could potentially introduce subtle bugs where we treat an HTTP status error the same as an API error which is likely not correct.